src/borg/archive.py

```diff
@@ -106,16 +107,16 @@ def unpack_many(self, ids, filter=None, preload=False):
         unpacker = msgpack.Unpacker(use_list=False)
         for _, data in self.fetch_many(ids):
             unpacker.feed(data)
-            items = [decode_dict(item, ITEM_TEXT_KEYS) for item in unpacker]
+            items = [Item(internal_dict=item) for item in unpacker]
```

Comment:
the internal dict of Item is holding data in the format that can be given to msgpack (and thus is also fed with data received from msgpack).